File: client/src/app/models/order.ts
export interface Order {
    id: number
    buyerEmail: string
    shippingAddress: ShippingAddress
    orderDate: string
    orderItems: OrderItem[]
    subtotal: number
    deliveryFee: number
    discount: number
    total: number
    orderStatus: string
    paymentSummary: PaymentSummary
  }
  
  export interface ShippingAddress {
    name: string
    line1: string
    line2?: string | null
    city: string
    state: string
    postal_code: string
    country: string
  }
  
  export interface OrderItem {
    productId: number
    name: string
    pictureUrl: string
    price: number
    quantity: number
  }
  
  export interface PaymentSummary {
    last4: number | string
    brand: string
    exp_month: number
    exp_year: number
  }
  
  export interface CreateOrder {
    shippingAddress: ShippingAddress
    paymentSummary: PaymentSummary
  }

File: API/Entities/OrderAggregate/Order.cs
using System;

namespace API.Entities.OrderAggregate;

public class Order
{
    public int Id { get; set; }
    public required string BuyerEmail { get; set; }
    public required ShippingAddress ShippingAddress { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public List<OrderItem> OrderItems { get; set; } = [];
    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }
    public long Discount { get; set; }
    public required string PaymentIntentId { get; set; }
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    public required PaymentSummary PaymentSummary { get; set; }

    public long GetTotal()
    {
        return Subtotal + DeliveryFee - Discount;
    }
}
